x-pack/plugins/lens/public/xy_visualization/axis_settings_popover.tsx

```diff
@@ -236,6 +286,178 @@ export const AxisSettingsPopover: React.FunctionComponent<AxisSettingsPopoverPro
+                    defaultMessage: 'Full',
+                  }),
+                  'data-test-subj': 'lnsXY_axisExtent_groups_full',
+                },
+                {
+                  id: `${idPrefix}dataBounds`,
+                  label: i18n.translate('xpack.lens.xyChart.axisExtent.dataBounds', {
+                    defaultMessage: 'Data bounds',
+                  }),
+                  'data-test-subj': 'lnsXY_axisExtent_groups_DataBounds',
+                  isDisabled: hasBarOrAreaOnAxis,
+                },
+                {
+                  id: `${idPrefix}custom`,
+                  label: i18n.translate('xpack.lens.xyChart.axisExtent.custom', {
+                    defaultMessage: 'Custom',
+                  }),
+                  'data-test-subj': 'lnsXY_axisExtent_groups_custom',
+                },
+              ]}
+              idSelected={`${idPrefix}${
+                hasBarOrAreaOnAxis && localExtent.mode === 'dataBounds' ? 'full' : localExtent.mode
+              }`}
+              onChange={(id) => {
+                const newMode = id.replace(idPrefix, '') as AxisExtentConfig['mode'];
+                setLocalExtent({
+                  ...localExtent,
+                  mode: newMode,
+                  lowerBound:
+                    newMode === 'custom' && dataBounds ? Math.min(0, dataBounds.min) : undefined,
+                  upperBound: newMode === 'custom' && dataBounds ? dataBounds.max : undefined,
+                });
+              }}
+            />
+          </EuiFormRow>
+          {localExtent.mode === 'custom' && (
+            <>
+              <EuiSpacer size="s" />
+              <EuiFlexGroup gutterSize="s">
+                <EuiFlexItem>
+                  <EuiFormRow
+                    display="rowCompressed"
+                    fullWidth
+                    label={i18n.translate('xpack.lens.xyChart.lowerBoundLabel', {
+                      defaultMessage: 'Lower bound',
+                    })}
+                    isInvalid={inclusiveZeroError || boundaryError}
+                    helpText={
+                      hasBarOrAreaOnAxis && !inclusiveZeroError
+                        ? i18n.translate('xpack.lens.xyChart.inclusiveZero', {
+                            defaultMessage:
+                              'Bounds have to include include zero for area and bar series',
+                          })
+                        : undefined
+                    }
+                    error={
+                      hasBarOrAreaOnAxis && inclusiveZeroError
+                        ? i18n.translate('xpack.lens.xyChart.inclusiveZero', {
+                            defaultMessage:
+                              'Bounds have to include include zero for area and bar series',
+                          })
+                        : undefined
+                    }
+                  >
+                    <EuiFieldNumber
```

Comment:
`EuiFieldNumber` needs a `compressed` prop in order to apply the compressed styles.